scripts/corelli/calibration/database.py

```diff
@@ -36,64 +37,165 @@ def has_valid_columns(table:TableWorkspace)->bool:
+def filename_bank_table( bankID:int, database_path:str, date:str, table_type:str = 'calibration' ) -> str:
     """
     Function that returns the absolute path for a bank file using corelli format:
     database_path/bank0ID/type_corelli_bank0ID_YYYYMMDD.nxs
     :param bankID bank number that is calibrated
+    :param date format YYYYMMDD
     :param database_path location of the corelli database (absolute or relative)
            Example: database/corelli/ for
                     database/corelli/bank001/
                     database/corelli/bank002/
     :param type 'calibration', 'mask' or 'acceptance'
     """
+
+    verify_date_YYYYMMDD('filename_bank_table', date)
     subdirectory:str = database_path + '/' + 'bank' + str(bankID).zfill(3)
     pathlib.Path(subdirectory).mkdir(parents=True, exist_ok=True)
     message = f'{"Cannot process Corelli filename_bank_table, table_type must be calibration, mask or acceptance"}'
     assert table_type == 'calibration' or table_type == 'mask' or table_type == 'acceptance', message
 
-    time:str = datetime.now().strftime('%Y%m%d') # format YYYYMMDD
-    filename = subdirectory + '/' + table_type + '_corelli_bank' + str(bankID).zfill(3) + '_' + time + '.nxs.h5'
+    filename = subdirectory + '/' + table_type + '_corelli_bank' + str(bankID).zfill(3) + '_' + date + '.nxs.h5'
     # make it portable
     filename = str(pathlib.Path(filename).resolve())
     return filename
 
 
-def save_bank_table( data:Workspace, bankID:int, database_path:str, table_type:str = 'calibration' ) -> None:
+def save_manifest_file( bankIDs:list, database_path:str, date:str, iso_timestamps:list)->None:
+    """
+    Function that saves or updates an existing manifest_corelli_date.csv file.
+    There is one file stored for each calibration day. If file exist it will append two columns:
+    bankID, timestamp (ISO format)
+    :param bankIDs input of bankIDs that have been calibrated
+    :param database_path location of the corelli database (absolute or relative)
+           Example: database/corelli/ for manifest file:
+                    database/corelli/manifest_corelli_YYYYMMDD.csv
+    :param date format YYYYMMDD
+    :param iso_timestamps list of timestamps for each calibrated bank in iso format YYYY-MM-DDTHH:MM:SS
+                          from datetime.datetime.now().replace(microsecond=0).isoformat()
+    """
+    # verify banks and timestamps are the same length
+    if len(bankIDs) != len(iso_timestamps):
+        raise ValueError('bankIDs and iso_timestamps list inputs must have the same lengths')
+
+    # verify timestamps are correct
+    for iso_timestamp in iso_timestamps:
+        if not re.match('[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}', iso_timestamp):
+            raise ValueError('iso_timestamp entry: ' + str(iso_timestamp) + ' is not valid iso format YYYY-MM-DDTHH:MM:SS')
+
+    filename = database_path + '/manifest_corelli_' + date + '.csv'
```

Comment:
since files of the format `somename_bankID_YYMMDD.xxx` are being stored in subdirectory `bankID`, would you consider files of the format `somename_corelli_YYMMDD.xxx` being stored in subdirectory `corelli`?
Also, in the future we may have to create subdirectories for the orientation-calibration, so having every file under some subdirectory now would help future migrations.

Reply:
OK, sounds good.